cmd/cp-main.go

```diff
@@ -317,6 +321,15 @@ func doCopySession(session *sessionV8) error {
 			// Save totalSize.
 			cpURLs.TotalSize = session.Header.TotalBytes
 
+			// Save storage class metadata if present
+			if cpURLs.TargetContent.Metadata == nil {
+				cpURLs.TargetContent.Metadata = make(map[string]string)
+			}
+
+			if _, ok := session.Header.CommandStringFlags["storage-class"]; ok {
```

Comment:
The check for the existence of `cpURLs.TargetContent.Metadata` can be moved inside the `if` block, that way we do not create an empty map in certain cases.